test/gomega/matchers/have_protocol_test.go

```diff
@@ -0,0 +1,28 @@
+package matchers_test
+
+import (
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/gloo/test/gomega/matchers"
+)
+
+var _ = Describe("HaveProtocol", func() {
+
+	It("ignores if there is no protocol specified", func() {
+		httpResponse := &http.Response{
+			Proto: matchers.HTTP1Protocol,
+		}
+		Expect(httpResponse).To(matchers.HaveProtocol(""))
+	})
+
+	It("matches the specified protocol", func() {
+		httpResponse := &http.Response{
+			Proto: matchers.HTTP1Protocol,
+		}
+		Expect(httpResponse).To(matchers.HaveProtocol(matchers.HTTP1Protocol))
+		Expect(httpResponse).To(matchers.HaveHTTP1Protocol())
+		Expect(httpResponse).ToNot(matchers.HaveHTTP2Protocol())
```

Comment:
nit: These two assertions feel redundant to me. I'm not sure how having both improves our certainty of the behavior of this utility. Can you help me understand?

Reply:
The aim is to prove a negative test case